feature/platform/transceiver/otg_tests/zr_temperature_test/zr_temperature_test.go

```diff
@@ -0,0 +1,226 @@
+// limitations under the License.
+
+package zr_temperature_test
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+        "github.com/openconfig/featureprofiles/internal/fptest"
+        "github.com/openconfig/ondatra"
+        "github.com/openconfig/ondatra/gnmi"
+        "github.com/openconfig/ondatra/gnmi/oc"
+        "github.com/openconfig/ygnmi/ygnmi"
+        "github.com/openconfig/ygot/ygot"
+	"github.com/golang/ygot/ygot/ygot"
+	"github.com/openconfig/featureprofiles/internal/deviations/deviations"
+	"github.com/openconfig/featureprofiles/internal/samplestream/samplestream"
+)
+
+const (
+	sensorType        = oc.PlatformTypes_OPENCONFIG_HARDWARE_COMPONENT_SENSOR
+	dp16QAM           = 1
+	targetOutputPower = -10
+	frequency         = 193100000
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// Topology:
+//
+//	dut:port1 <--> port2:dut
+
+func interfaceConfig(t *testing.T, dut1 *ondatra.DUTDevice, dp *ondatra.Port) {
+	d := &oc.Root{}
+	i := d.GetOrCreateInterface(dp.Name())
+	i.Enabled = ygot.Bool(true)
+	i.Type = oc.IETFInterfaces_InterfaceType_ethernetCsmacd
+	gnmi.Replace(t, dut1, gnmi.OC().Interface(dp.Name()).Config(), i)
+	// transceiverName := gnmi.Get(t, dut1, gnmi.OC().Interface(dp.Name()).Transceiver().State())
+	// OCcomponent := transceiverName + "-Optical0"
+	OCcomponent := opticalChannelFromPort(t, dut1, dp)
+	gnmi.Replace(t, dut1, gnmi.OC().Component(OCcomponent).OpticalChannel().Config(), &oc.Component_OpticalChannel{
+		TargetOutputPower: ygot.Float64(targetOutputPower),
+		Frequency:         ygot.Uint64(frequency),
+	})
+}
+func verifyTemperatureSensorValue(t *testing.T, dut1 *ondatra.DUTDevice, pStream *samplestream.SampleStream[float64], sensorName string) float64 {
+	temperatureSample := pStream.Next(t)
+	if temperatureSample == nil {
+		t.Fatalf("Temperature telemetry %s is not present", sensorName)
+	}
+	temperatureVal, ok := temperatureSample.Val()
+	if !ok {
+		t.Fatalf("Temperature %q telemetry is not present", temperatureSample)
+	}
+	// Check temperature return value of correct type
+	if reflect.TypeOf(temperatureVal).Kind() != reflect.Float64 {
+		t.Fatalf("Return value is not type float64")
+	} else if temperatureVal <= 0 && temperatureVal >= 300 {
+		t.Fatalf("The variable temperature instent is not between 0 and 300")
+	}
+	t.Logf("Temperature sample value %s: %v", sensorName, temperatureVal)
+	return temperatureVal
+}
+
+func TestZRTemperatureState(t *testing.T) {
+	dut1 := ondatra.DUT(t, "dut")
+	dp1 := dut1.Port(t, "port1")
+	dp2 := dut1.Port(t, "port2")
+	t.Logf("dut1: %v", dut1)
+	t.Logf("dut1 dp1 name: %v", dp1.Name())
+	intUpdateTime := 2 * time.Minute
+	interfaceConfig(t, dut1, dp1)
+	interfaceConfig(t, dut1, dp2)
+	gnmi.Await(t, dut1, gnmi.OC().Interface(dp1.Name()).OperStatus().State(), intUpdateTime, oc.Interface_OperStatus_UP)
+	transceiverName := gnmi.Get(t, dut1, gnmi.OC().Interface(dp1.Name()).Transceiver().State())
```

Comment:
is transceiverName really all of the transceiver state and not the name? If so, perhaps this var could be "transceiverState" ?

Reply:
This return transceiver component name same code snippet suggested in review comment from LimeHat.